### PR TITLE
Add filter by startswith, endswith and icontains to API

### PR DESCRIPTION
**Pull request by szok**
Added:
Filter by: startswith, endswith... to Charfield
Filter by: lte, gte ... to IntegerField
Filter by:  year, month to DateField
Filter by: lte, gte .. to DecimalField

### CODE REVIEW

File: src/ralph/api/viewsets.py [hunk 1]
```diff
@@ -122,6 +124,20 @@ class RalphAPIViewSet(
     metaclass=RalphAPIViewSetMetaclass
 ):
     extend_filter_fields = None
+    # allow_lookups = default_lookups.keys()
+    allow_lookups = {
+        models.IntegerField: [
```
Comment (mkurek) [inline src/ralph/api/viewsets.py]: imo it should be set, not list

File: src/ralph/api/viewsets.py [hunk 2]
```diff
@@ -150,8 +166,26 @@ def get_queryset(self):
...
+            field = field_name.split('__')
+            model_field_name = "__".join(field[:-1])
+            try:
+                model_field = get_field_by_relation_path(
+                    queryset.model, model_field_name
+                )
+            except FieldDoesNotExist:
+                continue
+
+            lookups = []
```
Comment (mkurek) [inline src/ralph/api/viewsets.py]: use set here too

File: src/ralph/api/viewsets.py [hunk 3]
```diff
@@ -150,8 +168,26 @@ def get_queryset(self):
...
             if value:
-                q_param = Q()
+                q_param = models.Q()
                 for field_name in field_filters:
-                    q_param |= Q(**{field_name: value})
+                    q_param |= models.Q(**{field_name: value})
                 queryset = queryset.filter(q_param)
+
+        for field_name, value in self.request.query_params.items():
+            field = field_name.split('__')
```
Comment (mkurek) [inline src/ralph/api/viewsets.py]: use `rpartition` here: https://docs.python.org/3.4/library/stdtypes.html#str.rpartition

File: src/ralph/tests/factories.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import factory
 from django.contrib.auth import get_user_model
 
-from ralph.tests.models import Manufacturer
+from ralph.tests.models import Bar, Manufacturer
```
Comment (ghost) [inline src/ralph/tests/factories.py]: Is `Bar` used somehow?

File: src/ralph/api/tests/test_viewsets.py [hunk 5]
```diff
@@ -87,6 +115,65 @@ def test_extend_filter_fields(self):
...
+
+    def test_query_filters_decimalfield(self):
+        request = self.request_factory.get('/api/bar')
+        bvs = BarViewSet()
+        request.query_params = {'price__gte': 20}
+        bvs.request = request
+        self.assertEqual(len(bvs.get_queryset()), 2)
+
+        # Failed filter
+        request.query_params = {'price__istartswith': 10}
```
Comment (xliiv) [inline src/ralph/api/tests/test_viewsets.py]: `price__istartswith` is this mistake?

**Comment by ghost** [comment]
:ok:

**Comment by mkurek** [comment]
:ok:

**Pull request merged**